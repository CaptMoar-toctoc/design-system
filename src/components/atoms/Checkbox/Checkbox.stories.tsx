import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Checkbox from './Checkbox';

export default {
	title: 'TocToc-DS/Atoms',
	component: Checkbox,
	argTypes: {
		checked:{control: 'boolean'}
	},
} as ComponentMeta<typeof Checkbox>;

const Template: ComponentStory<typeof Checkbox> = (args) => <Checkbox {...args} />;

export const CheckboxTT = Template.bind({});
CheckboxTT.args = {
	texto: 'Mi checkbox',
	disabled: false,	
	defaultChecked: false,
	handlerFn: () => { console.log(`click`) },
	customStyleObject: {},
};
