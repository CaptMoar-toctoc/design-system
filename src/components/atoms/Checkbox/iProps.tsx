export interface iProps {
	id?: string,
	checked?: boolean,
    handlerFn?: any,
    texto?: string,
    disabled?: boolean,
    customStyleObject?: object,
	defaultChecked?:boolean,
	value?:any
}