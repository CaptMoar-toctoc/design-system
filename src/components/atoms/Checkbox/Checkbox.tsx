import React, { FC } from 'react';
import { iProps } from './iProps';

import { Checkbox as CheckboxMUI, ThemeProvider } from '@mui/material';
import FormControlLabel from '@mui/material/FormControlLabel';

import theme from '../../theme/theme';

const Checkbox: FC<iProps> = ({
	id,
	checked,
    handlerFn,
    texto,
    disabled,
    customStyleObject,
	defaultChecked,
	value
}) => {
	return (
		<ThemeProvider theme={theme}>

			<FormControlLabel control={

				<CheckboxMUI
					id={id}
					onChange={handlerFn}
					disabled={disabled}
					sx={customStyleObject}
					defaultChecked={defaultChecked}
					checked={checked}
					value={value}
				/>

			} label={texto} />

		</ThemeProvider>
	);
}

export default Checkbox;