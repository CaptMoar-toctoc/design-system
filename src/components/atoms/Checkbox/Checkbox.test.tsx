import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import Checkbox from "./Checkbox";

describe('Checkbox Testing Render', () => {
	//inicializacion de props componente
	//aca van todos los props requerido y OPCIONALES clea
	const props = {
		texto: 'heeeey',
		handlerFn: () => { },
		disabled: true,
		defaultChecked: true,
	}

	//esto crea una imagen del componente renderizado
	test("Render SnapShot", () => {
		expect(render(<Checkbox {...props} />)).toMatchSnapshot();
	})

	//Validar que el componente sea un checkbox
	test("Validar que sea checkbox", () => {
		render(<Checkbox {...props} />)
		expect(screen.getByRole('checkbox')).toBeInTheDocument();
	})

	//esto validara que el texto sea el mismo que el que le entregasmos en el objeto props
	test("Validar que el texto sea correcto", () => {
		render(<Checkbox {...props} />)
		expect(screen.getByText(props.texto)).toBeInTheDocument();
	})

	//este campo valida que el campo checkbox este deshabilitado
	test("Validar que el componente este deshabilitado", () => {
		render(<Checkbox {...props} />)
		expect(screen.getByRole('checkbox')).toBeDisabled();
	})

	// Esto valida que cuando el campo disable cambie en los props, este se active, por defecto en los props esta deshabilitado
	test("Validar que el componente este habilitado", () => {
		render(<Checkbox {...props} disabled={false}/>)
		expect(screen.getByRole('checkbox')).not.toBeDisabled();
	})

	// valida que el campo check este habilitado
	test("Validar que el checkbox este checkeado", () => {
		render(<Checkbox {...props} />)
		expect(screen.getByRole('checkbox')).toBeChecked();
	})

	// valida que el checkbox no este checkeado en caso de cambiar los props 
	test("Validar que el checkbox no este checkeado", () => {
		render(<Checkbox {...props} defaultChecked={false}/>)
		expect(screen.getByRole('checkbox')).not.toBeChecked();
	})

	//valida console log, no estoy muy seguro de esto pero algo asi deveria ser
	// test("Validar que el checkbox haga un console log hola", () => {
	// 	render(<Checkbox {...props} />)
	// 	screen.getByRole('checkbox').click()
	// })
})

