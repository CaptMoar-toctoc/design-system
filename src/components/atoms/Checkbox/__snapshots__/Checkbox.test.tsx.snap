// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Testing Render Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          aria-disabled="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked Mui-disabled css-1a7yyzq-MuiButtonBase-root-MuiCheckbox-root"
          tabindex="-1"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input css-gb48lp"
            data-indeterminate="false"
            disabled=""
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-10a3z0g-MuiTypography-root"
        >
          heeeey
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        aria-disabled="true"
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked Mui-disabled css-1a7yyzq-MuiButtonBase-root-MuiCheckbox-root"
        tabindex="-1"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-gb48lp"
          data-indeterminate="false"
          disabled=""
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-10a3z0g-MuiTypography-root"
      >
        heeeey
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
